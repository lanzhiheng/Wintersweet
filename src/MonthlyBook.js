import React, { Component } from 'react';
import BookEntity from './BookEntity';
import './style/MonthlyBook.css';


class MonthlyBook extends Component {

  render() {
    const books = this.props.books;
    const bookList = books.map((x) => {
        return (<BookEntity key={x.name} book={x}/>);
    });
    return (
      <div className="monthly-book">
      <h4>{this.props.month}</h4>
      {bookList}
      </div>
    )
  }
}

export default MonthlyBook;
