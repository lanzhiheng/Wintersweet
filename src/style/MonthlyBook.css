.monthly-book h4{
  font-size: 30px;
  padding: 5px;
  text-align: left;
  color: green;
}
