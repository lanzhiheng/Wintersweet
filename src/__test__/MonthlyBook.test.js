import React from 'react';
import MonthlyBook from '../MonthlyBook';
import renderer from 'react-test-renderer';
const books = require('../data/books-test.json');

test('MonthlyBook display', () => {
  const component = renderer.create(
    <MonthlyBook month={books[0]['month']} books={books[0]['books']}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // 是否具有某个类
  expect(tree.props.className).toMatch(/\bmonthly-book\b/);
});
