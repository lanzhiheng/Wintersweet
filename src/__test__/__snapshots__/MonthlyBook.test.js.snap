exports[`test MonthlyBook display 1`] = `
<div
  className="monthly-book">
  <h4>
    8月份书单
  </h4>
  <div
    className="book-entity">
    <strong>
      《
      专业主义
      》
    </strong>
    <strong>
      大前研一
    </strong>
      
    <strong>
      done
    </strong>
  </div>
  <div
    className="book-entity">
    <strong>
      《
      不在独自旅行
      》
    </strong>
    <strong>
      大低&小低
    </strong>
      
    <strong>
      done
    </strong>
  </div>
  <div
    className="book-entity">
    <strong>
      《
      Atom Flight Manual
      》
    </strong>
    <strong>
      atom官方教程
    </strong>
      
    <strong>
      done
    </strong>
  </div>
</div>
`;
